diff --git a/lib/manager/events_manager.js b/lib/manager/events_manager.js
index a794d36..6f640a0 100644
--- a/lib/manager/events_manager.js
+++ b/lib/manager/events_manager.js
@@ -86,6 +86,7 @@ class EventsManager {
         }
         // Event hasn't been referenced in the current GTFS-RT feed
         else {
+            this.logger.debug('Not referenced in current GTFS-RT feed');
             let index = this.previousEvents[agency].indexOf(existingEvent);
             this.previousEvents[agency][index]['referenced'] = false; // Needed for clean up
         }
@@ -144,10 +145,11 @@ class EventsManager {
             this.logger.debug(`Publication of ${pendingEventsForAgencyCounter} pending events (${timestamp.toISOString()}.jsonld) successfully for agency: ${agency}`);
 
             // Clean up non-referenced events in previous events list
-            for(let e=0; e < this.previousEvents[agency]; e++) {
-                if(!this.previousEvents[agency][e]['connection']['referenced']) {
-                    this.previousEvents[agency].splice(this.previousEvents[agency].indexOf(e), 1);
+            for(let e=0; e < this.previousEvents[agency].length; e++) {
+		this.logger.debug('Event referenced?' + this.previousEvents[agency][e]['referenced']) 
+                if(this.previousEvents[agency][e]['referenced'] == false) {
                     this.logger.debug('Removed previous events (non-referenced):' + this.previousEvents[agency][e]['connection']['@id']);
+                    this.previousEvents[agency].splice(this.previousEvents[agency].indexOf(e), 1);
                 }
             }
         }
diff --git a/package.json b/package.json
index f78dc38..53799e7 100644
--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
     "follow-redirects": "^1.5.10",
     "gtfs-realtime-bindings": "~0.0.4",
     "gtfs2lc": "git+https://github.com/linkedconnections/gtfs2lc.git#development",
-    "gtfsrt2lc": "^0.2.1",
+    "gtfsrt2lc": "git+https://github.com/linkedconnections/gtfsrt2lc.git#development",
     "jsonld-stream": "^1.0.3",
     "level": "^2.1.2",
     "md5": "^2.2.1",
